### DISCUSSION TITLE
Welcome to MSFS2020-PilotPathRecorder Discussions!

### DISCUSSION THREAD

**Discussion by SAHorowitz** [SAHorowitz/MSFS2020-PilotPathRecorder#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Welcome to the PilotPathRecorder discussion section.  Thank you for the questions and contributions in advance.

**Comment by noodnik2** [comment]
Thanks for this app!  

I tried it today, and was really pleased how easy it was to get working.  My use case was simply to fly around my neighborhood, and awe in seeing everything from above.  Of course, given the amazing visuals already available in MSFS2020, some views were less spectacular in Google Maps than in the new MS sim, but generally I think this is an awesome idea and I plan take a look at the source a bit later to help me get a better feeling for how it works.  

It appears the filename used to store the path is the same for Web or App.  Forgive me / advise if this is not correct.  I wanted to save my path in both formats (without needing copy/rename in order to keep both kinds) so that I could try both out.  If this is indeed so, and you agree, I'd be happy to create an Issue, and propose a fix in a PR.

Nice work!

Regards,

-- Marty

**Reply by SAHorowitz** [reply]
Hi Marty. Thanks for the kind words. Glad you like the app. As you look at the source feel free to ask questions if you need or let me know any improvements areas you see. 

I am happy to add a "_web" or "_app" to the end of the file name. That would solve it. Thoughts?

**Reply by noodnik2** [reply]
I'd have the same urge for any possible combination of parameters that would affect the output (such as `Speed up first person playback`).  Consider adding (suffix of) the current timestamp to the suggested output filename?  In any case, if I could alter the suggested output filename it'd probably be good enough.  

Also, I thought it might be nice to have the chosen parameter configuration represented in the `description` tag of the output (for posterity, 😇).

BTW, I tried to set it up for development, but got stuck on two things:

1. Manifest signing - something about "Click Once" that I couldn't figure out; but, after disabling the "manifest" stuff altogether, I then got stuck on:
1. The same problem with the "SimConnect" problem that I see on the zines other people struggling with.  I tried the versions of both the "managed" and standard DLLs from the SDK, but it still complained about "incorrect format" when trying to load them.

Any suggestions appreciated, as I'd like to keep trying!

Regards,

-- Marty

P.S. - I haven't used Visual Studio for years, so I just installed VS19 (Community), pointed it at the`.sln` file, and expected everything to _"just work."_  Pretty naiive, eh?  In any case, hopefully I won't be the last that loads this project into an IDE, so I'll be happy to document the additional considerations / dependencies likely needing to be considered by newbies (e.g., thought to add a `README-dev.md` file linked from the top), so long as I can understand and get past them.

**Reply by SAHorowitz** [reply]
You were correct to disable the Manifest signing.  The other error was my fault.  I was referencing the Microsoft.FlightSimulator.SimConnect.dll from a directory outside the project.  I have moved that dll into the project and updated the project. Please re-download the code and try again.

**Reply by noodnik2** [reply]
Thanks!

I've been having some fun with a new "Live Camera" view I've added to [my fork](https://github.com/noodnik2/MSFS2020-PilotPathRecorder) - come take a look and tell me what you think! 

The familiarity I've gained with KML has me some ideas for some other Google Earth applications as well.

Regards,

-- Marty

**Reply by SAHorowitz** [reply]
Awesome.  Let me take a look over the next few days and get back to you.

**Comment by Richard-Walter** [comment]
Thanks for this recorder mate.  One suggestion would be to automate the sim connection functionality, so we don't have to manually do it.  E.g. code but in python: 

    # Init SimConnect, Wait for MSFS to respond
    try:
        sm = SimConnect()
    except:
        x=True
        print("WAITING FOR MSFS 2020 TO START")
        y=1
        while x == True:
            if y % 10 == 0:
                print("...")
            if y % 100 == 0:
                print("Did you Start MSFS 2020?")    
            try:
                sm = SimConnect()
                x=False   
            except:
                y=y+1
                continue
            time.sleep(2)

**Reply by SAHorowitz** [reply]
Richard -- thanks for the suggestion.  I will look into having a timer attempt to connect to sim every x seconds if currently disconnected.

**Comment by ccartter** [comment]
I am a MSFS user and would love to add this functionality. I have zero coding experience however and am new to github. I have followed your install instructions and I can't get the FS2020PlanePath app to launch. I have the correct version of dotnet and have launched the MSFS app before trying to get yours to launch, as instructed. Just downloaded it today, so I have the latest version of your app. Any advice for this newbie? Thanks, Chris

**Reply by SAHorowitz** [reply]
Hey Chris. Sorry you are running into trouble. What error are you getting?  Does the program start and not record or does rhe app not start at all?

**Comment by ccartter** [comment]
It's weird. It doesn't start at all. I'm running Windows 10 Home with .Net
5.0 runtime v5.0.2 x64

any suggestions? I'm new to Windows so spell it out for me!

thanks
Chris

**Reply by SAHorowitz** [reply]
Hmm.. That is strange. Just to be sure you unzipped the PilotPathRecorder zip into a directory and start the exe? You can't run it from within the zip file. If so then please take a screenshot of the PilotPathRecorder files (in detail view) so I can check a few things. Also what language is your Windows set to? I mean English, French etc.

**Comment by noodnik2** [comment]
Question form the sidelines: how do you produce that .zip file?  Are you
using GitHub Actions, or??

I'm looking to create my first "release," and would like to do something
similar.

Thanks!

**Reply by SAHorowitz** [reply]
I produce the zip by using the 'build | publish' command in the IDE.  Then I run the installer and install it on my computer.  Then I find the installed files and zip them up (minus the DB file). I then uninstall the app to be ready for the next time I want to produce a zip file.

**Comment by ccartter** [comment]
English. See attached for directory.

thanks
Chris

**Reply by SAHorowitz** [reply]
Sorry I don't see any attachments in your reply.  I am looking for something like this.  And you should be launching the app by starting FS2020PlanePath.exe

![image](https://user-images.githubusercontent.com/60077862/107304200-9c777580-6a4e-11eb-8e4b-4d987c2f0c82.png)

**Comment by ccartter** [comment]
weird.Here it is again

[image: Path recorder.PNG]

**Reply by SAHorowitz** [reply]
Sorry. Still not seeing it.  Paste the image directly into the message.  And verify you are starting the program by double-clicking on FS2020PlanePath.exe

**Comment by ccartter** [comment]
here's a jpeg. I am opening the correct .exe file.

[image: pilotpath.jpg]

**Reply by SAHorowitz** [reply]
Chris, 
 I'm sorry your images are still not coming thru.  Let's try this.  Does your directory list look like what I pasted in my message above?  At least 33 files, same filenames, file sizes, etc?  Also, you don't get any dialog box or error when you start the FS2020PilotPath.exe?   One other thing to try - right click on the exe and pick 'Run as an Administrator' and see if that fixes it. 

Thanks.
Steve

**Comment by ccartter** [comment]
All files are identical to yours. Run as Admin did the trick. Thanks for
hanging in there with me. I'll let you know how it goes.

best regards
Chris

**Reply by SAHorowitz** [reply]
Happy to hear.  Hopefully, the app works for you without issues.  Good luck and let us know if you need more help.

**Comment by amei-2222** [comment]
I am enjoying your generosity in producing this valuable utility.  I do have one question/request.  Is there an easy way to convert a MSFS 2020 *.FLN file to a *.KML file?  I just want to preview how the flight plan would look on google earth.  Any suggestions?

Thanks!

**Reply by SAHorowitz** [reply]
Hi amei-2222.  I did find a way to do what you are wanting.  Hopefully, you use LittleNavMap.  If not, try it out it is a great program and what I use to plan my flights.  Anyways, in LittleNavMap make a flight plan and then choose File and then Export Flight Plan in Other Formats.  Chose GPX format.  Save the file off somewhere.  Then go to https://gpx2kml.com/ and choose that GPX file you saved off and have it convert it to KML.  Download the KML and use it in Google Earth.  Then you should be good to go.

I tried it out and it worked fine for me.  Let me know how it goes for you.

Steve

**Reply by amei-2222** [reply]
Thanks for your response.

That doesn't quite solve my problem.  I do use LittleNavMap.  It has its own ability to use Google maps internal to LNM with an addon so I can observe the Google map details internal to LittleNavMap.  But that is not the problem I have.  Unfortunately, a plan developed in LittleNavMap and saved as a .FLN file for use by MSFS 2020 does not render as the same flight plan when you bring up the Navlog in MSFS 2020 (that is, some of the waypoints are developed differently somehow by MSFS 2020).  For some reason MSFS 2020 picks some different waypoints.  Very Frustrating!  What I am trying to do is examine the .fln file after it is saved out of MSFS (the only option is to save it as a .flt file again out of MSFS 2020 which I have no way of converting to a .kml file).  So, that is my dilemma.

Thanks,
